### PR TITLE
IGNITE-16550-4

### PR DESCRIPTION
**Pull request by denis-chudov**

### CODE REVIEW

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java [hunk 1]
```diff
@@ -542,11 +456,14 @@ private void createTableLocally(long causalityToken, String name, UUID tblId, in
             throw new IgniteInternalException(IgniteStringFormatter.format("Cannot create a table [name={}, id={}]", name, tblId), th);
         });
 
-        CompletableFuture.allOf(tablesByIdVv.get(causalityToken), tablesVv.get(causalityToken)).thenRun(() -> {
+        // TODO should be reworked in IGNITE-16763
+        schemaFut.thenCompose(v -> tablesByIdVv.get(causalityToken)).thenRun(() -> {
```
Comment (vldpyatkov) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:460]: Why does the composed future required?

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java [hunk 2]
```diff
@@ -498,39 +442,9 @@ private void createTableLocally(long causalityToken, String name, UUID tblId, in
...
-            }
-        });
-
-        var table = new TableImpl(internalTable, schemaRegistry);
-
-        tablesVv.update(causalityToken, previous -> {
-            var val = new HashMap<>(previous);
-
-            val.put(name, table);
+        var table = new TableImpl(internalTable, null);
```
Comment (vldpyatkov) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:445]: It is the main place in source where we are creating a table, hence it looks weird that we use a constructor with null value.

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java [hunk 3]
```diff
@@ -549,11 +466,14 @@ private void createTableLocally(long causalityToken, String name, UUID tblId, in
             return completedFuture(val);
         });
 
-        CompletableFuture.allOf(tablesByIdVv.get(causalityToken), tablesVv.get(causalityToken)).thenRun(() -> {
+        // TODO should be reworked in IGNITE-16763
+        schemaFut.thenCompose(v -> tablesByIdVv.get(causalityToken)).thenRun(() -> {
```
Comment (vldpyatkov) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:470]: You can to hold this on tablesByIdVv.

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java [hunk 4]
```diff
@@ -247,7 +266,7 @@ private CompletableFuture<?> onTableCreate(ConfigurationNotificationEvent<TableV
         }
 
         try {
-            createTableLocally(
+            return createTableLocally(
```
Comment (vldpyatkov) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:269]: I think, there is no a reason to do this anymore.

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java [hunk 5]
```diff
@@ -189,20 +191,18 @@ public TableManager(
         };
         clusterNodeResolver = topologyService::getByAddress;
 
-        tablesVv = new VersionedValue<>(registry, HashMap::new);
-        tablesByIdVv = new VersionedValue<>(registry, HashMap::new);
+        tablesByIdVv = new VersionedValue<>(null, HashMap::new);
+
+        this.schemaManager.listen(SchemaEvent.COMPLETE, (parameters, e) -> {
+            tablesByIdVv.complete(parameters.causalityToken());
```
[the remark this answers is not in the document]
Comment (denis-chudov) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:197]: `SchemaEvent.COMPLETE` is fired when SchemaManager#registriesVv completes, it means it is fired no sooner than the revision update happens.

File: modules/schema/src/main/java/org/apache/ignite/internal/schema/SchemaManager.java [hunk 6]
```diff
@@ -0,0 +1,374 @@
...
+
+        UUID tblId = tblCfg.id().value();
+
+        String tableName = tblCfg.name().value();
+
+        SchemaDescriptor schemaDescriptor = SchemaSerializerImpl.INSTANCE.deserialize((schemasCtx.newValue().schema()));
+
+        CompletableFuture<?> res = createSchema(causalityToken, tblId, tableName, schemaDescriptor);
+
+        fireEvent(SchemaEvent.CREATE, new SchemaEventParameters(causalityToken, tblId, schemaDescriptor), null);
```
[the remark this answers is not in the document]
Comment (denis-chudov) [reply modules/schema/src/main/java/org/apache/ignite/internal/schema/SchemaManager.java:116]: fixed

**Pull request merged**